src/jobTree/jobStores/awsJobStore.py

```diff
@@ -413,16 +430,25 @@ def _newFileID( self, sharedFileName=None ):
         else:
             return str( uuid.uuid5( self.sharedFileJobID, sharedFileName ) )
 
-    def _getFileVersion( self, jobStoreFileID ):
+    def _getFileVersionAndBucket(self, jobStoreFileID):
         """
-        :rtype: str
+        :rtype: tuple(str version, AWS bucket)
         """
         for attempt in retry_sdb( ):
             with attempt:
                 item = self.versions.get_attributes( item_name=jobStoreFileID,
-                                                     attribute_name='version',
+                                                     attribute_name=['version','bucketName'],
                                                      consistent_read=True )
-        return item.get( 'version', None )
+        bucketName = item.get('bucketName', None)
+        return (None,None) if bucketName is None else ( item.get('version', None), getattr(self, bucketName))
+
+    def _getFileVersion( self, jobStoreFileID ):
+        version,bucket = self._getFileVersionAndBucket(jobStoreFileID)
+        if bucket is None:
+            assert version is None
+        else:
+            assert bucket is self.files
```

Comment:
I'm not sure this is a good assignment to always do. In cases like writeSharedFileStream, getFileVersion can be called on files that don't yet exist, causing this assertion to fail.